install/threesdk/lib/SDKContainers.py

```diff
@@ -6,20 +14,78 @@ def __init__(self, core, args):
-        if not identity and self.args.identity:
-            return self.args.identity
-        if not identity:
-            identity = self.core.IT.Tools.ask_string("what is your threebot name (identity)?")
-        if "." not in identity:
-            identity += ".3bot"
-        identity = identity.lower()
-        if self.args.identity != identity:
-            self.args.identity = identity
-            self.args.words = None
-            self.args.email = None
-
-        return identity
+    def _get_user(self):
+        resp = requests.get("https://{}/explorer/users".format(self.args.explorer), params={"name": self.args.identity})
+        if resp.status_code == 404:
+            return None
+        else:
+            users = resp.json()
+            if users:
+                return users[0]
+            return None
+
+    def _identity_ask(self, identity=None, explorer=None):
+        def _fill_identity_args(identity, explorer):
+            def fill_words():
+                words = self.core.IT.Tools.ask_string("what are the words associated with your identity?")
+                self.args.words = words
+
+            if identity:
+                if self.args.identity != identity:
+                    self.args.reset()
+                self.args.identity = identity
+
+            if explorer:
+                self.args.explorer = explorer
+            elif not self.args.explorer:
+                response = self.core.IT.Tools.ask_choices(
+                    "Which network would you like to register to? ", ["mainnet", "testnet", "devnet", "none"]
+                )
+                self.args.explorer = NETWORKS.get(response, None)
+            if not self.args.explorer:
+                return True
+            if not self.args.identity:
+                identity = self.core.IT.Tools.ask_string("what is your threebot name (identity)?")
+                if "." not in identity:
+                    identity += ".3bot"
+                self.args.identity = identity
+            user = self._get_user()
+            if not user and not self.args.email:
+                email = self.core.IT.Tools.ask_string("what is your email associated with your identity?")
+                self.args.email = email
+            else:
+                print("Configured email for this identity is {}".format(user["email"]))
+                self.args.email = user["email"]
+            if not self.args.words:
+                if user:
+                    fill_words()
+                else:
+                    self.args.words = None
+            if user and self.args.words:
+                # time to do validation of words
+                while True:
+                    try:
+                        seed = self.core.IT.Tools.to_entropy(self.args.words, english.words)
+                    except Exception:
+                        choice = self.core.IT.Tools.ask_choices("Invalid words where given what would you like to do?", ["restart", "reenter"])
```

Comment:
Should be more explicit about the error. What does invalid means ? Why ? and what is the advice to get it right ?

Reply:
here we can have the message say:

"Seems one or more more words entered is invalid. What would you like to do?" with 
"restart initialization of identity" and "reenter phrase" as options